lib/src/ctx/context.rs

```diff
@@ -91,6 +100,31 @@ impl<'a> Context<'a> {
 		}
 	}
 
+	pub fn with_live_sess(mut self, session: &Session) -> Self {
+		self.add_value(SESSION_KEY, Value::None);
+		match session.sd.clone() {
```

Comment:
I was curious how clone benefits here than just borrowing the Session parameter?

Something like
```
match &session.sd{
}

or 

if let Some(sd) = &session.sd {
        self.add_value(AUTH_KEY, sd.clone());
} else {
        self.add_value(AUTH_KEY, Value::Null);
}
```

P.S Rust noob here, please correct me if it is some pattern I am not aware of

Reply:
I think the only added benefit is in case of null, it won't clone

Reply:
You are right @arriqaaq, we should delay cloning as much as possible so the branches that don't need an owned copy don't incur an unnecessary cost.

Reply:
Created https://github.com/surrealdb/surrealdb/issues/2829 for this

Reply:
@phughk How about using

```rust
match &session.sd {
    Some(sd) => self.add_value(AUTH_KEY, sd.clone()),
    None => self.add_value(AUTH_KEY, Value::Null),
}
```

as per @arriqaaq's suggestion?